src/orange/univ/UniverseVisitor.hh

```diff
@@ -0,0 +1,85 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/univ/UniverseVisitor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/math/Algorithms.hh"
+#include "orange/OrangeData.hh"
+
+#include "UniverseTypeTraits.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Apply a functor to a simple or rect-array universe
```

Comment:
```suggestion
 * Apply a functor to a universe tracker of unknown type.
```

Reply:
I realize now there's an inconsistency between our "tracker" name and the "universe" here. I think universe is the abstraction of a unit, rectangular array; and a tracker is the helper class that does the actual navigation/tracking/state update. Just like we have `SurfaceTypeTraits` but `LocalSurfaceVisitor`, maybe rename this class `TrackerVisitor`. Also please add a small unit test.